apps/prairielearn/src/sync/course-db.ts

```diff
@@ -891,6 +891,32 @@ function checkAllowAccessDates(rule: { startDate?: string; endDate?: string }):
   };
 }
 
+/**
+ * It seems to be relatively common for instructors to accidentally put multiple
+ * UIDs in the same string, like "uid1@example.com, uid2@example.com". While we
+ * are pretty loose in what we accept as UIDs, they should never contain commas or
+ * whitespace, so we'll warn about that.
+ */
+function checkAllowAccessUids(rule: { uids?: string[] }): string[] {
+  const warnings: string[] = [];
+
+  const uidsWithWhitespace = (rule.uids ?? []).filter((uid) => /\s/.test(uid));
+  if (uidsWithWhitespace.length > 0) {
+    warnings.push(
+      `The following access rule UIDs contain unexpected whitespace: ${formatValues(uidsWithWhitespace)}`,
+    );
+  }
+
+  const uidsWithCommas = (rule.uids ?? []).filter((uid) => /,/.test(uid));
+  if (uidsWithCommas.length > 0) {
+    warnings.push(
+      `The following access rule UIDs contain unexpected commas: ${formatValues(uidsWithCommas)}`,
+    );
+  }
```

Comment:
Originally I was going to put just a single warning:

> The following access rule UIDs appear to be invalid: ...

However, I assumed someone scanning that might not immediately pick up on the distinction between the commas separating UIDs in our output and the commas they'd accidentally embedded in their own UIDs string, so I split them up so we could prime people to look for either whitespace or commas.